trin-core/src/portalnet/overlay.rs

```diff
@@ -187,6 +188,44 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
             .collect()
     }
 
+    /// Returns an iterator over all the entries in the routing table.
+    pub fn table_entries(&self) -> Vec<(NodeId, Enr, NodeStatus)> {
+        self.kbuckets
+            .write()
+            .iter()
+            .map(|entry| {
+                (
+                    *entry.node.key.preimage(),
+                    entry.node.value.enr().clone(),
+                    entry.status,
+                )
+            })
+            .collect()
+    }
+
+    /// Returns vector of all ENR node IDs of nodes currently contained in the routing table mapped to JSON Value.
+    pub fn routing_table_info(&self) -> Value {
+        let buckets: Vec<(String, String, String)> = self
+            .table_entries()
+            .iter()
+            .map(|(node_id, enr, node_status)| {
+                (
+                    node_id.to_string(),
+                    enr.to_base64(),
+                    format!("{:?}", node_status.state),
+                )
+            })
+            .collect();
+
+        json!(
```

Comment:
Agreed w/ ^, I feel like the json conversion should be handled in `trin-history/src/jsonrpc.rs`